docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,338 @@
+* An optional `gardenlet` section that contains:
+    * Certain aspects of the `gardenlet` deployment configuration, such as the number of replicas, the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+    * The `GardenletConfiguration` resource that contains controllers configuration, feature gates, etc.
+    
+Either the `seed` or the `gardenlet` section must be specified, but not both: 
+
+* If the `seed` section is specified, `gardenlet` should not be deployed in the shoot, and a new seed should be registered based on the `seed` section.
+* If the `gardenlet` section is specified, `gardenlet` should be deployed in the shoot, and it will register a new seed upon startup based on the `seedConfig` section of the `GardenletConfiguration` resource.
+
+A ManagedSeed allows fine-tuning the seed and the `gardenlet` configuration of shooted seeds in order to deviate from the global defaults, e.g. lower the concurrent sync for some of the seed's controllers or enable a feature gate only on certain seeds. Also, it simplifies the deletion protection of such seeds. 
+
+Also, the `ManagedSeed` resource is a more powerful alternative to the `use-as-seed` annotation. The implementation of the `use-as-seed` annotation itself could be refactored to use a `ManagedSeed` resource extracted from the annotation by a controller.
+    
+Although in this proposal a ManagedSeed is always a "shooted seed", that is a Shoot that is registered as a Seed, this idea could be further extended in the future by adding a `type` field that could be either `Shoot` (implied in this proposal), or something different. Such an extension would allow to register and manage as Seed a cluster that is not a Shoot, e.g. a GKE cluster.
+
+Last but not least, ManagedSeeds could be used as the basis for creating and deleting seeds automatically via the `ManagedSeedSet` resource that is described in [ManagedSeedSets](#managedseedsets).
+
+Unlike the `Seed` resource, the `ManagedSeed` resource is namespaced. If created in the `garden` namespace, the resulting seed is globally available. If created in a project namespace, the resulting seed can be used as a "private seed" by shoots in the project, either by being decorated with project-specific taints and labels, or by being of the special `PrivateSeed` kind that is also namespaced. The concept of private seeds / cloudprofiles is described in [Issue #2874](https://github.com/gardener/gardener/issues/2874). Until this concept is implemented, `ManagedSeed` resources might need to be restricted to the `garden` namespace, similarly to how shoots with the `use-as-seed` annotation currently are.
+
+Example `ManagedSeed` resource with a `seed` section:
+
+```yaml
+apiVersion: seedmanagement.gardener.cloud/v1alpha1
+kind: ManagedSeed
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  shootName: crazy-botany # Shoot that should be registered as a Seed
```

Comment:
Maybe one small suggestion:
Instead of having one field `shootName` we could also use a nested structure like
```yaml
spec:
  shoot:
    name: crazy-botany
```
This way we would be completely flexible to add new fields later on without breaking API-compatibility. 
Though, I don't have a use case in mind as of now, but it might become useful :) 
WDYT?

Reply:
Done